autoload/test/strategy.vim

```diff
@@ -75,8 +75,14 @@ function! test#strategy#neovim(cmd) abort
   let term_position = get(g:, 'test#neovim#term_position', 'botright')
   execute term_position . ' new'
   call termopen(a:cmd)
-  au BufDelete <buffer> wincmd p " switch back to last window
-  startinsert
+  au BufDelete <buffer> wincmd p " switch back to last window on delete
+
+  " Start insert mode only when running a specific test
+  if stridx(a:cmd, '--filter') != -1
```

Comment:
This logic is runner-specific, so it doesn't belong in a strategy. I understand the need for it, but we have to find another way.

Reply:
Oh yeah that's true. Forgot about the other runners. Is there a way to differentiate the command that was ran? Like detect if it's `:TestNearest`?

Reply:
Typically you'd set different CLI options depending on whether it was `:TestNearest`, `:TestFile` or `:TestSuite`, but in this case we want different _strategy_ options, and we don't yet have a concept of "strategy options" that you could pass via the command.

At the moment strategies don't have the knowledge whether `:TestNearest` was used, they receive only the test command. I'm not sure whether giving strategies this knowledge is the correct thing to do, but maybe it would make sense.

An alternative that's currently possible is to use different strategies for different granularity. That would mean we'd need to have two different strategies, and then the user can configure which one they want for which granularity.

Reply:
Okay guess we can leave it like the way it used to be until there's a concept of strategy options. Not sure if I'm the only one who needs the no auto focus update anyways, so I'll just use these changes locally.